ludwig/encoders/date_encoders.py

```diff
@@ -232,34 +253,33 @@ def __init__(
             default_weights_regularizer=weights_regularizer,
             default_bias_regularizer=bias_regularizer,
             default_activity_regularizer=activity_regularizer,
-            # default_weights_constraint=weights_constraint,
-            # default_bias_constraint=bias_constraint,
             default_norm=norm,
             default_norm_params=norm_params,
             default_activation=activation,
             default_dropout=dropout,
         )
 
-    def call(
+    def forward(
             self,
-            inputs,
-            training=None,
-            mask=None
-    ):
+            inputs: torch.Tensor,
+            training: Optional[bool] = None,
+            mask: Optional[bool] = None
+    ) -> Dict[str, torch.Tensor]:
         """
-            :param input_vector: The input vector fed into the encoder.
-                   Shape: [batch x 19], type tf.int8
-            :type input_vector: Tensor
-            :param training: bool specifying if in training mode (important for dropout)
+            :param inputs: The input vector fed into the encoder.
+                Shape: [batch x DATE_INPUT_SIZE], type torch.int8
+            :type inputs: Tensor
+            :param training: bool specifying if in training mode (important for
+                dropout).
             :type training: bool
-            :param mask: bool specifying masked values
+            :param mask: bool specifying masked values.
             :type mask: bool
          """
         # ================ Embeddings ================
-        input_vector = tf.cast(inputs, tf.int32)
+        input_vector = inputs.type(torch.int32)
 
         scaled_year = self.year_fc(
-            tf.cast(input_vector[:, 0:1], tf.float32),
+            input_vector[:, 0:1].type(torch.float32),
```

Comment:
Question about this -- from the `input_shape` property and the `DATE_INPUT_SIZE` definition, my understanding was that the year would be the first 4 indices of the input feature, where each index is a number between 0-9. So to get the input for `self.year_fc`, you'd need `input_vector[:, 0:4]` followed by converting the list of ints to a single integer. Similarly for other inputs to the month, day, etc. encoders.

I'm not sure if I'm missing some context tho or if my understanding of the input representation is incorrect?

Reply:
If the size 19 inputs are broken down like: (YYYY)(MM)(DD)(W)(YDYD)(HH)(MM)(SS), then we should certainly pass in 0:4 for year, 4:6 for month, etc.

I've pinged the ludwig channel to confirm, since the previous embedding code seems to be working off of input of size 9.

Reply:
Changed to 9. Based on offline discussion, 19 is likely copied from the H3 encoder code. date_features' returns are size 9 for (year, month, day, weekday, yearday, hours, minutes, seconds, seconds of day). I've also updated the encoder unit tests as well.